phase1-coordinator/src/coordinator_state.rs

```diff
@@ -2517,6 +2527,45 @@ impl CoordinatorState {
             .collect()
     }
 
+    /// This will drop a participant if it hasn't accepted a task for more than
+    /// [crate::environment::Environment]'s `participant_acceptance_timeout`.
+    fn update_participant_accepted_drops(
+        &mut self,
+        time: &dyn TimeSource,
+    ) -> Result<Vec<DropParticipant>, CoordinatorError> {
+        let participant_acceptance_timeout = self.environment.participant_acceptance_timeout();
+        let now = time.utc_now();
+
+        self.current_contributors
+            .clone()
+            .iter()
+            .chain(self.current_verifiers.clone().iter())
+            .filter_map(|(participant, participant_info)| {
+                // Ensure we get the proper elapsed time since the last task was assigned.
+                // If no tasks have been assigned yet, we measure from the starting time.
+                let last_seen = participant_info.last_released.unwrap_or(participant_info.first_seen);
```

Comment:
I'm understanding correctly that last released is the time since the participant last released a lock? There are some situations where a participant is not performing any work, such as when it is waiting for a verifier to complete chunks before it can take a lock on a chunk. Another situation is if the contributor has completed all its contributions, but it is waiting for the round to complete. If this takes longer than this timeout it will cause this contributor to drop?

Reply:
We already have a lock timeout, I think it partially overlaps with this one currently.

Reply:
That's correct - this is why I postulated that this timeout may have to be quite large, possibly larger than the `last_seen_timeout`. However, it may be worth it to update that logic to be a bit more intelligent as well, if there's any way to distinguish in a participant's state if it's waiting on someone else or if it actually has work to pick up, then decisions can be made a bit more granularly. It'll be complex to get that right though, but I can give it some thought for sure. What do you think?

Reply:
I think we may need to associate the timing with tasks rather than chunk locks to achieve the desired effect.

Reply:
Perhaps for any of the participant's currently available tasks, how long since they were assigned. I just can't remember if tasks can be assigned but not available to be worked on...

Reply:
Other than this key issue, the pull request looks good, hopefully we can address this without too much trouble

Reply:
> Perhaps for any of the participant's currently available tasks, how long since they were assigned

it seems to have the same behaviour when

> the contributor has completed all its contributions, but it is waiting for the round to complete

Reply:
> Perhaps for any of the participant's currently available tasks, how long since they were assigned.

Wouldn't this coincide with the lock timeout though? It made more sense to me initially to start counting from when they joined, or last finished. It may help however to disallow dropping participants based on this timeout if the list of `assigned_tasks` is empty or something, for sure.

Reply:
> That's correct - this is why I postulated that this timeout may have to be quite large, possibly larger than the last_seen_timeout. 

If it needs to be so large, perhaps we could consider having a timeout on the round itself instead? It's simple and the effect of that is easy to reason about.

> Wouldn't this coincide with the lock timeout though?

Locking a task doesn't precisely match when a participant was assigned a task (when the task is in the participant's queue from my understanding) it's usually a subset of that. One issue though with the approach I mentioned is that I can't remember whether say a contributor can be assigned a task in its queue before it is allowed to work on it (before the task has been verified).

Reply:
> It'll be complex to get that right though, but I can give it some thought for sure. What do you think?

Yes I think getting this right in the way I had originally considered will be a bit complex, this is making me think more about whether a round timeout may be better.

Reply:
> If it needs to be so large, perhaps we could consider having a timeout on the round itself instead? It's simple and the effect of that is easy to reason about.

We could, though having this timeout helps us find replacement contributors/verifiers in some circumstances, so we can be more fault-tolerant. Unless we restart the round anyways in the case of replacing contributors and verifiers?

> Locking a task doesn't precisely match when a participant was assigned a task (when the task is in the participant's queue from my understanding) it's usually a subset of that.

When you say assigned here, you mean the assigning that happens at the beginning of the round right (tasks going into the `assigned_tasks` queue)? This could work too, but we would need to restart the timeout once an assigned task was completed, I think. Otherwise you are just running a timer for the round. It seemed most logical to me at the time to use the release of a lock for refreshing that timeout.

Maybe a round timeout would at least be a bit simpler, but depending on what actions are performed when replacements are used, we may have a lengthier ceremony.

Reply:
If it doesn't fail the integration tests, then I guess this could be okay to include in its current form. 

I'm still a bit hesitant about putting it in there for this universal setup because it's conceivable that it would drop someone close to the end of their 10 hour contribution just because one of our verifiers is going slow. Perhaps we can add a way to control the timeout from the config file, so it will be easy to adjust this in production if required?

I can't remember for sure whether 1. the assigned_tasks queue is full from the beginning (requiring the contributor to poll every task until it becomes available) or whether it 2. gets filled up as tasks become available after having been verified. Either way we could conceivably create a system that tracks when tasks effectively become available to a contributor who is not currently working on any tasks (they have no locks), and drops the contributor when they don't start working on those tasks. That was the original intention for this issue. At the moment as I understand it, this will drop the contributor when they haven't been working on a task regardless of whether there were any tasks actually available for them to work on?

Reply:
What happens in the situation where the contributor is waiting a long time for another contributor who is very slow? Exaggerated effect of this in the universal setup.

Reply:
As discussed on the standup, you raised some very valid points. The PR has been set back to draft and will be shelved for the time being, since I have other priorities at the moment. Will ping when I've come up with a new design.